addons/mrp_subcontracting_purchase/tests/test_mrp_subcontracting_purchase.py

```diff
@@ -901,3 +901,46 @@ def test_return_subcontracted_product_to_supplier_location(self):
         return_picking = self.env['stock.picking'].browse(return_picking_id)
         return_picking.button_validate()
         self.assertEqual(return_picking.state, 'done')
+
+    @freeze_time('2000-05-05')
+    def test_global_visibility_days_affect_lead_time(self):
+        """ Don't count global visibility days more than once, make sure a PO generated from
+        replenishment/orderpoint has a sensible planned reception date.
+        """
+        try:
+            self.env['sale.order']
+        except KeyError:
+            self.skipTest('This test requires the installation of the following modules: sale')
```

Comment:
`try-except` is rarely elegant
For such case, we often do
```py
if 'sale.order' not in self:
    self.skipTest
```

That being said, I don't get why you want to process a SO here
If really you need an out, you can just confirm a delivery picking